device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -160,8 +163,9 @@ else if (attr.startsWith(SHARED_ACCESS_TOKEN_ATTRIBUTE))
         }
 
         initIotHubClient(hostname, deviceId, sharedAccessKey, sharedAccessToken, protocol);
-        this.logger = new CustomLogger(this.getClass());
-        logger.LogInfo("DeviceClient object is created successfully, method name is %s ", logger.getMethodName());
+        
+        logger.info("DeviceClient object is created successfully, method name is {} ", 
+        		Thread.currentThread().getStackTrace()[2].getMethodName());
```

Comment:
Please avoid hard coding values. Use private static final instead. 
Again another reason to isolate as a method this in another class and lessen the code footprint.